src/server/shell/accessibility_manager.cpp

```diff
@@ -114,12 +117,24 @@ struct MouseKeysTransformer: public mir::input::InputEventTransformer::Transform
 
 mir::shell::AccessibilityManager::AccessibilityManager(
     std::shared_ptr<mir::options::Option> const& options,
-    std::shared_ptr<input::InputEventTransformer> const& event_transformer) :
+    std::shared_ptr<input::InputEventTransformer> const& event_transformer,
+    std::shared_ptr<mir::graphics::Cursor> const& cursor) :
     enable_key_repeat{options->get<bool>(options::enable_key_repeat_opt)},
     enable_mouse_keys{options->get<bool>(options::enable_mouse_keys_opt)},
     event_transformer{event_transformer},
-    transformer{std::make_shared<MouseKeysTransformer>()}
+    transformer{std::make_shared<MouseKeysTransformer>()},
+    cursor{cursor}
 {
+    auto const cursor_scale = static_cast<float>(options->get<double>(mir::options::cursor_scale_override_opt));
```

Comment:
I know it would need an insane scale to be problematic, but this is a narrowing conversion of external input. The value should be range checked or clamped. (I'd clamp to something much stricter: (0..100] say.)